tests/FunctionalTestCase.php

```diff
@@ -625,4 +657,31 @@ private function isFailCommandEnabled(): bool
 
         return isset($document->enableTestCommands) && $document->enableTestCommands === true;
     }
+
+    /**
+     * Checks whether the specified file exists in the system PATH and is executable
+     *
+     * @param string $filename
+     * @return bool
+     */
+    private function isExecutablePathenv(string $filename): bool
+    {
+        if (is_executable($filename)) {
+            return true;
+        }
+
+        if ($filename !== basename($filename)) {
+            return false;
```

Comment:
What case is this covering?

Per my earlier comments, I think this would be better relocated to a private method as `ClientSideEncryptionSpecTest::skipIfMongocryptdIsUnavailable()`.

```
protected function skipIfMongocryptdIsUnavailable(): void
{
    $paths = explode(PATH_SEPARATOR, getenv("PATH"));

    foreach ($paths as $path) {
        if (is_executable($path . DIRECTORY_SEPARATOR . 'mongocryptd')) {
            return;
        }
    }

    $this->markTestSkipped('mongocryptd binary is not in PATH');
}
```

Then, that method can be called from `ClientSideEncryptionSpecTest::setUp()`. Even though not every test within the CSFLE test suite might actually require a running `mongocryptd`, I think that's a reasonable approach to avoid the errors from [PHPLIB-647](https://jira.mongodb.org/browse/PHPLIB-647) that we're trying to avoid here.

You can also consider using `markTestIncomplete` instead, since that's noisier than merely skipping the test. The goal here is the avoid local test failures while alerting the developer to the fact that they may be missing binaries -- so in that sense an incomplete test may be more likely to show up in PHPUnit logs and alert them to the issue.